scipy/stats/tests/test_multivariate.py

```diff
@@ -309,6 +346,79 @@ def test_frozen(self):
 
+    def _assert_outputs_close(self, x, d1, d2, d2_args):
+        assert_allclose(d1.entropy(), d2.entropy(**d2_args))
+        assert_allclose(d1.pdf(x), d2.pdf(x, **d2_args))
+        assert_allclose(d1.logpdf(x), d2.logpdf(x, **d2_args))
+        assert_allclose(d1.cdf(x), d2.cdf(x, **d2_args))
+        assert_allclose(d1.logcdf(x), d2.logcdf(x, **d2_args))
+
+    def test_outputs_with_inverse_cov(self):
+        # Test method outputs for a range of inputs when `inverse_cov` is
+        # specified.
+        np.random.seed(1234)
+        # In a five-dimensional space:
+        x1, mean1 = np.random.randn(5), np.random.randn(5)
+        cov1 = np.abs(np.random.randn(5))
+        inverse_cov1 = 1 / cov1
+        # Evaluate on a matrix to test broadcasting in high dimensions.
+        z1 = np.random.randn(2, 3, len(mean1))
+        # In one-dimension:
+        x2, mean2, cov2, inverse_cov2 = 3.4, 0.1, 5.5, 1 / 5.5
+        # Evaluate on a vector to test broadcasting in one dimension.
+        z2 = np.random.randn(2)
+        args = [(x1, mean1, cov1, inverse_cov1),
+                (z1, mean1, cov1, inverse_cov1),
+                (x2, mean2, cov2, inverse_cov2),
+                (z2, mean2, cov2, inverse_cov2)]
+        for x, m, cov, inv_cov in args:
+            self._assert_outputs_close(
+                x,
+                multivariate_normal(mean=m, cov=cov),
+                multivariate_normal,
+                dict(mean=m, cov=None, inverse_cov=inv_cov),
+            )
+
+    def test_frozen_with_inverse_cov(self):
+        # The frozen distribution should agree with the regular one when
+        # `inverse_cov` is set.
+        np.random.seed(1234)
+        dim = 5
+        x, mean = np.random.randn(dim), np.random.randn(dim)
+        cov = np.abs(np.random.randn(dim))
+        inv_cov = 1 / cov
+        self._assert_outputs_close(
+            x,
+            multivariate_normal(mean, cov),
+            multivariate_normal(mean, cov=None, inverse_cov=inv_cov),
+            {}
+        )
+
+    def test_prefers_cov_over_inverse_cov(self):
+        # Check `cov` is preferred when both `cov` and `inverse_cov` are
+        # passed.
+        np.random.seed(1234)
+        dim = 5
+        x, mean = np.random.randn(dim), np.random.randn(dim)
+        cov = np.abs(np.random.randn(dim))
+        inv_cov = 1 / cov
+        # Not the inverse of `cov``:
+        bad_inv_cov = 100 * inv_cov
+        # Check `bad_inv_cov` is ignored for non-frozen distributions.
+        self._assert_outputs_close(
+            x,
+            multivariate_normal(mean=mean, cov=cov),
+            multivariate_normal,
+            dict(mean=mean, cov=cov, inverse_cov=bad_inv_cov),
+        )
+        # Same again for frozen distribution.
+        self._assert_outputs_close(
+            x,
+            multivariate_normal(mean, cov),
+            multivariate_normal(mean, cov=cov, inverse_cov=bad_inv_cov),
+            {},
+        )
```

Comment:
We won't need this anymore when we're no longer adding `inverse_cov` an argument.